modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx

```diff
@@ -310,11 +282,70 @@ const SetupAnalyticsCloudPage = ({
 					);
 				}
 			}
+		};
+
+		try {
+			setIsLoadingSubmitButton(true);
+
+			if (featureFlags.includes('LPS-159127')) {
+				await actRaysourceContact(
+					removeContactRoleRaysource,
+					removeHighPriorityContact,
+					project,
+					sessionId,
+					provisioningServerAPI
+				);
+				await actRaysourceContact(
+					associateContactRoleRaysource,
+					addHighPriorityContact,
+					project,
+					sessionId,
+					provisioningServerAPI
+				);
+				await actLiferayContact(
+					addHighPriorityContact,
+					associateContactRoleLiferay,
+					project,
+					client
+				);
+				await actLiferayContact(
+					removeHighPriorityContact,
+					removeContactRoleLiferay,
+					project,
+					client
+				);
+			}
+
+			handleDataSubmit();
 			setIsLoadingSubmitButton(false);
 
 			handlePage(true);
-		} catch {
-			setIsLoadingSubmitButton(false);
+		} catch (error) {
+			if (error.cause === STATUS_CODE.conflict) {
+				try {
+					await actLiferayContact(
```

Comment:
why are we trying `actLiferayContact` again after the conflict error? can we reorder it in a way where we can improve these if statements and try catches?

Reply:
wasn't it decided that we would handle the request when the error was due to a conflict? That's why, when we're in this condition, we try actLiferayContact. The provisioning side already has the role, so we need to accept the role on the Liferay side.

Reply:
yeah you have the right idea, but is there a way to clean it up visually so we don't have to nest try/catches, call functions multiple times, set `setIsLoadingSubmitButton(false)` so many times, things like that. 
while something may be conceptually correct, it could still be optimized visually or implementation wise

Reply:
Okay, I thought I had misunderstood, I'll try to find a better way to run this process.

Reply:
@ryanschuhler Because there are setStates within all the try/catches, is it possible to abstract these out? From our backend perspective, we do want to limit duplicate code, and in order to do that, we usually pass through variables in methods. However, I'm unfamiliar as to the best practices for frontend. Please let me know if I'm not understanding this correctly, but I remember react training mentioned we shouldn't be passing around states unless absolutely necessary and they really should stay within their own component/files.

Is there any way to reduce code duplication on frontend code within this example on Pedro's PR? I did read up on ErrorBoundaries but even those don't really touch useStates if you're expecting certain frontend behavior changes.